lib/openstudio-standards/infiltration/infiltration.rb

```diff
@@ -0,0 +1,183 @@
+      boundary_condition = surface.outsideBoundaryCondition
+      unless boundary_condition == 'Outdoors' || boundary_condition == 'Ground'
+        return 0.0
+      end
+
+      # Per area infiltration rate for this surface
+      surface_type = surface.surfaceType
+      infil_rate_cfm_per_ft2 = nil
+      case boundary_condition
+      when 'Outdoors'
+        case surface_type
+        when 'RoofCeiling'
+          infil_rate_cfm_per_ft2 = component_infil_rates_cfm_per_ft2[type]['roof']
+        when 'Wall'
+          infil_rate_cfm_per_ft2 = component_infil_rates_cfm_per_ft2[type]['exterior_wall']
+        when 'Floor'
+          infil_rate_cfm_per_ft2 = component_infil_rates_cfm_per_ft2[type]['floor_over_unconditioned']
+        end
+      when 'Ground'
+        case surface_type
+        when 'Wall'
+          infil_rate_cfm_per_ft2 = component_infil_rates_cfm_per_ft2[type]['below_grade_wall']
+        when 'Floor'
+          infil_rate_cfm_per_ft2 = component_infil_rates_cfm_per_ft2[type]['slab_on_grade']
+        end
+      else
+        OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.Infiltration', "Could not determine infiltration surface type for #{surface.name}, defaulting to 0 component infiltration rate.")
+        return 0.0
+      end
+
+      # Area of the surface
+      area_m2 = surface.netArea
+      area_ft2 = OpenStudio.convert(area_m2, 'm^2', 'ft^2').get
+
+      # Rate for this surface
+      comp_infil_rate_cfm = area_ft2 * infil_rate_cfm_per_ft2
+      comp_infil_rate_m3_per_s = OpenStudio.convert(comp_infil_rate_cfm, 'cfm', 'm^3/s').get
+
+      return comp_infil_rate_m3_per_s
+    end
+
+    # @!endgroup Surface
+
+    # @!group SubSurface
+
+    # Determine the component infiltration rate for a sub surface
+    # Details described in Gowri, Krishnan, Winiarski, David W, and Jarnagin, Ronald E. Infiltration modeling guidelines for commercial building energy analysis. United States: N. p., 2009. Web. doi:10.2172/968203.
+    #
+    # @param sub_surface [OpenStudio::Model::SubSurface] OpenStudio SubSurface object
+    # @param type [String] choices are 'baseline' and 'advanced'
+    # @return [Double] component infiltration rate in m^3/s
+    def self.sub_surface_component_infiltration_rate(sub_surface, type: 'baseline')
+      # Define the envelope component infiltration rates
+      component_infil_rates_cfm_per_ft2 = {
+        'baseline' => {
+          'opaque_door' => 0.40,
+          'loading_dock_door' => 0.40,
+          'swinging_or_revolving_glass_door' => 1.0,
+          'vestibule' => 1.0,
+          'sliding_glass_door' => 0.40,
+          'window' => 0.40,
+          'skylight' => 0.40
+        },
+        'advanced' => {
+          'opaque_door' => 0.20,
+          'loading_dock_door' => 0.20,
+          'swinging_or_revolving_glass_door' => 1.0,
+          'vestibule' => 1.0,
+          'sliding_glass_door' => 0.20,
+          'window' => 0.20,
+          'skylight' => 0.20
```

Comment:
these are 0.4 in Table 1 of https://www.pnnl.gov/main/publications/external/technical_reports/PNNL-18898.pdf

Reply:
I updated the reference.  Seems like the values in the code came from Table 5.7 of Thornton, Brian A, Rosenberg, Michael I, Richman, Eric E, Wang, Weimin, Xie, YuLong, Zhang, Jian, Cho, Heejin, Mendon, Vrushali V, Athalye, Rahul A, and Liu, Bing. Achieving the 30% Goal: Energy and Cost Savings Analysis of ASHRAE Standard 90.1-2010. United States: N. p., 2011. Web. doi:10.2172/1015277.